code/modules/events/space_vines/vine_event.dm

```diff
@@ -8,22 +8,105 @@
 	description = "Kudzu begins to overtake the station. Might spawn man-traps."
 	min_wizard_trigger_potency = 4
 	max_wizard_trigger_potency = 7
+	admin_setup = list(
+		/datum/event_admin_setup/set_location/spacevine,
+		/datum/event_admin_setup/multiple_choice/spacevine,
+		/datum/event_admin_setup/input_number/spacevine_potency,
+		/datum/event_admin_setup/input_number/spacevine_production,
+	)
 
 /datum/round_event/spacevine
 	fakeable = FALSE
+	///Override location the vines will spawn in.
+	var/turf/override_turf
+	///used to confirm if admin selected mutations should be used or not.
+	var/mutations_overridden = FALSE
+	///Admin selected mutations that the kudzu will spawn with, can be set to none to act as mutationless kudzu.
+	var/list/override_mutations = list()
+	///Potency of the spawned kudzu.
+	var/potency
+	///Production value of the spawned kuduz.
+	var/production
 
 /datum/round_event/spacevine/start()
 	var/list/turfs = list() //list of all the empty floor turfs in the hallway areas
 
-	var/obj/structure/spacevine/vine = new()
 
-	for(var/area/station/hallway/area in GLOB.areas)
-		for(var/turf/floor as anything in area.get_contained_turfs())
-			if(floor.Enter(vine))
-				turfs += floor
+	if(override_turf)
+		turfs += override_turf
+	else
+		var/obj/structure/spacevine/vine = new()
 
-	qdel(vine)
+		for(var/area/station/hallway/area in GLOB.areas)
+			for(var/turf/floor as anything in area.get_contained_turfs())
```

Comment:
```suggestion
			for(var/turf/open/floor in area.get_contained_turfs())
```

otherwise they spawn in walls and that's weird imo... i see that it's like this on line 48 but i don't think it's worth it? unsure